JBPM-7514 - Create archetype for spring boot based service project

tsurdilo opened the pull request.

tsurdilo commented:
@mswiderski servie project spring boot archetype. See the README.md for instructions. If built by default the apps index page (after you auth with user/user) looks like this:

<img width="1359" alt="screen shot 2018-08-02 at 11 27 14 pm" src="https://user-images.githubusercontent.com/119422/43623277-40db67b6-9695-11e8-8535-7f59cd3d1718.png">

mswiderski commented on kie-archetypes/kie-service-spring-boot-archetype/src/main/resources/META-INF/archetype-post-generate.groovy:
```diff
@@ -0,0 +1,181 @@
+    println "- adding spring boot starter dependency";
+    pomContent = pomContent.replace(springBootStarterMarker, defaultSpringBootStarterDepends);
+
+    println "- adding hibernate 2 dependency";
```
I believe this should be H@ instead of hibernate 2

mswiderski commented on kie-archetypes/kie-service-spring-boot-archetype/src/main/resources/META-INF/archetype-post-generate.groovy:
```diff
@@ -0,0 +1,181 @@
+    println "- adding spring boot starter dependency";
+    pomContent = pomContent.replace(springBootStarterMarker, optimizerSpringBootStarterDepends);
+
+    println "- remove hibernate 2 dependency";
```
same here H2 instead of hibernate 2

mswiderski commented on kie-archetypes/kie-service-spring-boot-archetype/README.md:
```diff
@@ -0,0 +1,84 @@
+for this property:
+1. default: includes BRM, BPM, Case Management, BPM-UI, and DMN
+2. businessmanagement: includes BRM and DMN
+3. optimizer: includes BRM, BRP, and DMN
```
can we change the property name to be:
- bpm 
- brm
- planner
just os it's shorter and easier to grasp

mswiderski commented on kie-archetypes/kie-service-spring-boot-archetype/src/main/resources/archetype-resources/pom.xml:
```diff
@@ -0,0 +1,62 @@
+  <parent>
+    <groupId>org.springframework.boot</groupId>
+    <artifactId>spring-boot-starter-parent</artifactId>
+    <version>1.5.10.RELEASE</version>
```
can we upgrade to 1.5.14.RELEASE

mswiderski commented on kie-archetypes/kie-service-spring-boot-archetype/src/main/resources/archetype-resources/pom.xml:
```diff
@@ -0,0 +1,62 @@
+    <maven.compiler.source>1.8</maven.compiler.source>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+    <narayana.version>5.6.4.Final</narayana.version>
```
can we upgrade to 5.9.0.Final

mswiderski commented on kie-archetypes/kie-service-spring-boot-archetype/src/main/resources/archetype-resources/pom.xml:
```diff
@@ -0,0 +1,62 @@
+    KIE_SPRING_BOOT_STARTER_MARKER
+    <dependency>
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-starter-web</artifactId>
```
spring-boot-starter-web is already brought by any of kie server starters so no need to have it here

mswiderski commented on kie-archetypes/kie-service-spring-boot-archetype/src/main/resources/archetype-resources/pom.xml:
```diff
@@ -0,0 +1,62 @@
+    </dependency>
+    <dependency>
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-starter-thymeleaf</artifactId>
```
not sure we should have this dependency here... it might not be used by users and sort of enforcing thymeleaf out of the box does not sound to good for me

mswiderski commented on kie-archetypes/kie-service-spring-boot-archetype/src/main/resources/archetype-resources/pom.xml:
```diff
@@ -0,0 +1,62 @@
+    DATABASE_MARKER
+    <dependency>
+      <groupId>net.sourceforge.nekohtml</groupId>
+      <artifactId>nekohtml</artifactId>
```
same here, let's try to have as few dependencies as possible ... except those that we bring with kie server :)

mswiderski commented on kie-archetypes/kie-service-spring-boot-archetype/src/main/resources/archetype-resources/src/main/java/__artifactId__/DefaultController.java:
```diff
@@ -0,0 +1,19 @@
+import org.springframework.web.bind.annotation.RequestParam;
+
+@Controller
+public class DefaultController {
```
do we really need this? Can't we just have static html page that will be filtered on project creation and then we don't need this at all.

mswiderski commented on kie-archetypes/kie-service-spring-boot-archetype/src/main/resources/archetype-resources/src/main/resources/application.properties:
```diff
@@ -0,0 +1,22 @@
+server.port=${appServerPort}
+
+#thymeleaf configuration
+spring.thymeleaf.mode=LEGACYHTML5
```
let's try to avoid thymeleaf altogether

mswiderski commented on kie-archetypes/kie-service-spring-boot-archetype/src/main/resources/archetype-resources/src/main/resources/templates/index.html:
```diff
@@ -0,0 +1,43 @@
+            url: "/rest/server" ,
+            dataType: "xml" ,
+            success: function(xml) {
+                var capabilitiesList = xml.getElementsByTagName("capabilities");
```
maybe have the capabilities just based on the selected type of appType only - so we would nicely should it has capabilities of business automation or decision management or business optimisation

mswiderski commented on kie-archetypes/kie-service-spring-boot-archetype/src/main/resources/archetype-resources/src/main/resources/templates/index.html:
```diff
@@ -0,0 +1,43 @@
+<div class="jumbotron text-white font-weight-bold text-center" style="background: url(/images/bg.jpg) no-repeat center center fixed; -webkit-background-size: cover; -moz-background-size: cover; -o-background-size: cover; background-size: cover;">
+    <h1 class="display-4">Welcome to your new KIE Services App!</h1>
+
+    <p class="lead">App Server has successfully started with following capabilities:
```
Maybe just see that Service has successfully started with {} capabilities

mswiderski commented on kie-archetypes/kie-service-spring-boot-archetype/src/main/resources/archetype-resources/pom.xml:
```diff
@@ -0,0 +1,62 @@
+  </parent>
+
+  <properties>
+    <version.org.kie>${project.version}</version.org.kie>
```
can we make this as parameter with defaults to be project.version but allow people to give the version of KIE when generating it

mswiderski commented on kie-archetypes/kie-service-spring-boot-archetype/src/main/resources/archetype-resources/src/main/java/__artifactId__/DefaultController.java:
```diff
@@ -0,0 +1,19 @@
+package ${groupId}.${artifactId};
```
same here, package parameter instead of group and artifact - it can default then to just groupId

mswiderski commented:
@tsurdilo when I use this archetype it fails with following error:

command used to generate it:

```
mvn archetype:generate -DarchetypeGroupId=org.kie -DarchetypeArtifactId=kie-service-spring-boot-archetype -DarchetypeVersion=7.10.0-SNAPSHOT -DgroupId=com.company -DartifactId=test-service -Dversion=1.0-SNAPSHOT -Dpackage=com.company.service -DappType=default
```

```
[ERROR] Parser Exception: archetype-resources/src/main/resources/templates/index.html
[ERROR] org.apache.velocity.runtime.parser.ParseException: Encountered "{" at line 26, column 16.
Was expecting one of:
    <EOF>
    "(" ...
    <RPAREN> ...
    <ESCAPE_DIRECTIVE> ...
    <SET_DIRECTIVE> ...
    "##" ...
    "\\\\" ...
    "\\" ...
    <TEXT> ...
    "*#" ...
    "*#" ...
    <STRING_LITERAL> ...
    <IF_DIRECTIVE> ...
    <STOP_DIRECTIVE> ...
    <INTEGER_LITERAL> ...
    <FLOATING_POINT_LITERAL> ...
    <WORD> ...
    <BRACKETED_WORD> ...
    <IDENTIFIER> ...
    <DOT> ...
    "{" ...
    "}" ...

	at org.apache.velocity.runtime.parser.Parser.generateParseException(Parser.java:3360)
	at org.apache.velocity.runtime.parser.Parser.jj_consume_token(Parser.java:3237)
	at org.apache.velocity.runtime.parser.Parser.process(Parser.java:260)
	at org.apache.velocity.runtime.parser.Parser.parse(Parser.java:105)
	at org.apache.velocity.runtime.RuntimeInstance.parse(RuntimeInstance.java:1042)
	at org.apache.velocity.runtime.RuntimeInstance.parse(RuntimeInstance.java:972)
	at org.apache.velocity.Template.process(Template.java:120)
	at org.apache.velocity.runtime.resource.ResourceManagerImpl.loadResource(ResourceManagerImpl.java:415)
	at org.apache.velocity.runtime.resource.ResourceManagerImpl.getResource(ResourceManagerImpl.java:335)
	at org.apache.velocity.runtime.RuntimeInstance.getTemplate(RuntimeInstance.java:1102)
	at org.apache.velocity.app.VelocityEngine.mergeTemplate(VelocityEngine.java:498)
	at org.apache.maven.archetype.generator.DefaultFilesetArchetypeGenerator.processTemplate(DefaultFilesetArchetypeGenerator.java:735)
	at org.apache.maven.archetype.generator.DefaultFilesetArchetypeGenerator.processFileSet(DefaultFilesetArchetypeGenerator.java:540)
	at org.apache.maven.archetype.generator.DefaultFilesetArchetypeGenerator.processTemplates(DefaultFilesetArchetypeGenerator.java:785)
	at org.apache.maven.archetype.generator.DefaultFilesetArchetypeGenerator.processArchetypeTemplates(DefaultFilesetArchetypeGenerator.java:514)
	at org.apache.maven.archetype.generator.DefaultFilesetArchetypeGenerator.processFilesetProject(DefaultFilesetArchetypeGenerator.java:628)
	at org.apache.maven.archetype.generator.DefaultFilesetArchetypeGenerator.processFilesetModule(DefaultFilesetArchetypeGenerator.java:564)
	at org.apache.maven.archetype.generator.DefaultFilesetArchetypeGenerator.generateArchetype(DefaultFilesetArchetypeGenerator.java:203)
	at org.apache.maven.archetype.generator.DefaultArchetypeGenerator.processFileSetArchetype(DefaultArchetypeGenerator.java:129)
	at org.apache.maven.archetype.generator.DefaultArchetypeGenerator.generateArchetype(DefaultArchetypeGenerator.java:100)
	at org.apache.maven.archetype.generator.DefaultArchetypeGenerator.generateArchetype(DefaultArchetypeGenerator.java:143)
	at org.apache.maven.archetype.generator.DefaultArchetypeGenerator.generateArchetype(DefaultArchetypeGenerator.java:169)
	at org.apache.maven.archetype.DefaultArchetypeManager.generateProjectFromArchetype(DefaultArchetypeManager.java:73)
	at org.apache.maven.archetype.mojos.CreateProjectFromArchetypeMojo.execute(CreateProjectFromArchetypeMojo.java:207)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)

```

tsurdilo commented:
@mswiderski updated with all your suggestions - thanks!!
archetype metadata will not let me default package property to ${groupId} so I default it to org.kie. Hope that is ok. You can specify your own anyways when generating service app. 

Here is updated image of the index page (no thyleaf any more ;) )

<img width="1356" alt="sample" src="https://user-images.githubusercontent.com/119422/43652924-6f3b87ec-96fb-11e8-9ca4-6f60a8353ae9.png">

tsurdilo commented:
@mswiderski @MarianMacik hi :) we ok on this now?

mswiderski commented on kie-archetypes/kie-service-spring-boot-archetype/src/main/resources/archetype-resources/pom.xml:
```diff
@@ -0,0 +1,50 @@
+  </parent>
+
+  <properties>
+    <version.org.kie>${project.version}</version.org.kie>
```
this is still not parameterised so it will always be the same version as archetype - can be have another property that defaults to project version?

tsurdilo commented:
@mswiderski updated - you can set -DkieVersion=X.Y.X now. Default will be the project version. Also made the package property dynamic instead of previously hard-coded to org.kie, defaults to groupId. Updated README so users can know too

tsurdilo commented:
@mswiderski also added kie version on the landing page:

<img width="1363" alt="sample" src="https://user-images.githubusercontent.com/119422/43727642-66808446-9957-11e8-88c3-9443ec69a771.png">

mswiderski commented:
looks nice, just tried and the version is taken into account, great.

One thing that seems to be missing is the spring boot maven plugin in the generated pom.xml

```
<build>
      <plugins>
        <plugin>
          <groupId>org.springframework.boot</groupId>
          <artifactId>spring-boot-maven-plugin</artifactId>
        </plugin>
      </plugins>
    </build>
```

tsurdilo commented:
@mswiderski added

MarianMacik commented on kie-archetypes/kie-service-spring-boot-archetype/README.md:
````diff
@@ -0,0 +1,115 @@
+-Dpackage=YOUR_APP_PACKAGE
+```
+
+The default value of this property is org.kie
````
The default value is the groupId of the archetype.

tsurdilo commented:
@MarianMacik updated - thanks

tsurdilo commented:
@mswiderski aligned the landing page with that of bootstrapjbpm

<img width="1362" alt="sample" src="https://user-images.githubusercontent.com/119422/43784393-eb16bc5e-9a18-11e8-9736-4f034fc6306a.png">

mswiderski commented:
Thanks, could we also add these icons/cards?

Last thing I think we need here are the scripts to launch it - build and launch as we talked about it. Do you think we can add them (fine to have Unix for now and windows will follow)

tsurdilo commented:
@mswiderski sure ill add - do you mean the 3 icons below the generate button on the boostrapjbpm service app? Not sure I fully understand what icons - thanks!

mswiderski commented:
Yes, exactly these three

tsurdilo commented:
@mswiderski added launch.sh and launch.cmd scripts as well as a .mvn directory for the maven wrapper. This allows users to build and launch our application w/o having Maven installed on their machine. Please look at the updated README

after building the app from archetype all you should have to do is 
chmod 755 launch.sh
./launch.sh

it will download maven if not on machine and then run mvn clean spring-boot:run

Let me know what you think

MarianMacik commented on kie-archetypes/kie-service-spring-boot-archetype/src/main/resources/archetype-resources/launch.sh:
```diff
@@ -0,0 +1,287 @@
+
+if [ -z "$M2_HOME" ] ; then
+  ## resolve links - $0 may be a link to maven's home
+  PRG="$0"
```
So the script name can be the path to maven's home?

MarianMacik commented on kie-archetypes/kie-service-spring-boot-archetype/src/main/resources/archetype-resources/launch.sh:
```diff
@@ -0,0 +1,287 @@
+  M2_HOME=`cd "$M2_HOME" && pwd`
+
+  cd "$saveddir"
+  # echo Using m2 at $M2_HOME
```
Is this a debug echo or just comment?

MarianMacik commented on kie-archetypes/kie-service-spring-boot-archetype/src/main/resources/archetype-resources/launch.sh:
```diff
@@ -0,0 +1,287 @@
+  M2_HOME=`dirname "$PRG"`/..
+
+  # make it fully qualified
+  M2_HOME=`cd "$M2_HOME" && pwd`
```
Have you tried using `readlink -e $M2_HOME`? It should do the same.

MarianMacik commented on kie-archetypes/kie-service-spring-boot-archetype/src/main/resources/archetype-resources/launch.sh:
```diff
@@ -0,0 +1,287 @@
+    else
+      PRG="`dirname "$PRG"`/$link"
+    fi
+  done
```
Just a possible improvement - would the `readlink` utility solve this? (Maybe it is not available on Mac, I don't know.)

mswiderski commented:
@tsurdilo I think the script is a bit too much or an overkill - we should not make that too complex - at least for me this script is too complex to follow it while I believe we don't need that much complexity. What the script should do is:
- build all model projects in the parent folder - if any
- build all kjar projects in the parent folder - if any
- build the service (itself) project
- run the app after successful build

Maybe we could reuse [this script](https://github.com/BootstrapJBPM/jbpm-bootstrap-service/blob/master/launch.sh) as it pretty much does the job and it's easy to follow.
Not sure we should download anything from the script ... if maven is not there then it should be documented as prerequisite 

Overall I tested this briefly (not all options etc) and it looks correct. One thing we could look into is to reduce the size of the project as right now it's more than 1.5MB and that is mainly due to image and the css file. Maybe we can reduce quality of the image to make it way smaller, wdyt?

tsurdilo commented:
@mswiderski updated per your request. here are changes:

- reduced size: removed the background image, removed the large css file, page looks much simpler now but is very light
- added ability to change db types with param -DdbType and possible options being h2,myseq, and postgres (default h2)
- added application-dev.properties with default kieserver.controller configured (uses default value of -DjbpmConsolePort being 8080)
- removed the super huge build scripts and added lanuch.sh and launch-dev.sh. They rely on ${artifactId} and ${version} of the project generated to find itself and the corresponding -kjar -module and the final jar for running the app

mswiderski commented on kie-archetypes/kie-service-spring-boot-archetype/src/main/resources/META-INF/archetype-post-generate.groovy:
```diff
@@ -0,0 +1,357 @@
+spring.datasource.password=sa
+spring.datasource.url=jdbc:h2:./target/spring-boot-jbpm;MVCC=true
+spring.datasource.driver-class-name=org.h2.Driver
+spring.datasource.type=org.apache.tomcat.jdbc.pool.XADataSource
```
this data source type should be removed, it caused problems when moving to newer version of spring boot and it's not really needed

mswiderski commented on kie-archetypes/kie-service-spring-boot-archetype/src/main/resources/META-INF/archetype-post-generate.groovy:
```diff
@@ -0,0 +1,357 @@
+spring.datasource.password=jbpm
+spring.datasource.url=jdbc:mysql://localhost:3306/jbpm
+spring.datasource.driver-class-name=com.mysql.jdbc.jdbc2.optional.MysqlXADataSource
+spring.datasource.type=org.apache.tomcat.jdbc.pool.XADataSource
```
same here, remove the datasource.type

mswiderski commented on kie-archetypes/kie-service-spring-boot-archetype/src/main/resources/META-INF/archetype-post-generate.groovy:
```diff
@@ -0,0 +1,357 @@
+spring.datasource.password=jbpm
+spring.datasource.url=jdbc:postgresql://localhost:5432/jbpm
+spring.datasource.driver-class-name=org.postgresql.xa.PGXADataSource
+spring.datasource.type=org.apache.tomcat.jdbc.pool.XADataSource
```
same here, remove the datasource.type

mswiderski commented on kie-archetypes/kie-service-spring-boot-archetype/src/main/resources/META-INF/archetype-post-generate.groovy:
```diff
@@ -0,0 +1,357 @@
+
+    println "- adding database configuration";
+    println "   - using dbType: " + dbType;
+    if( dbType == "h2" ) {
```
I would actually include all the them in pom.xml as profiles so users:
- have the profiles in every application
- they have a complete set of information (driver, application.properties setup) in the app in case they would like to use another database
- less complexity in the archetype itself
- less options to chose from when generating app

mswiderski commented on kie-archetypes/kie-service-spring-boot-archetype/src/main/resources/META-INF/maven/archetype-metadata.xml:
```diff
@@ -0,0 +1,95 @@
+      <defaultValue>bpm</defaultValue>
+    </requiredProperty>
+    <!-- dbType options are h2,mysql,postgres -->
+    <requiredProperty key="dbType">
```
as mentioned, I'd removed it and generate maven profiles instead with duplicates of application.properties per data base - mysql and postgres

mswiderski commented on kie-archetypes/kie-service-spring-boot-archetype/src/main/resources/archetype-resources/src/main/java/DefaultWebSecurityConfig.java:
```diff
@@ -0,0 +1,29 @@
+        http
+        .csrf().disable()
+        .authorizeRequests()
+        .antMatchers("/*").authenticated()
```
maybe have security only for /rest/* so the landing page does not require authentication, wdyt?

mswiderski commented on kie-archetypes/kie-service-spring-boot-archetype/src/main/resources/archetype-resources/src/main/resources/application.properties:
```diff
@@ -0,0 +1,19 @@
+kieserver.serverId=${artifactId}
+kieserver.serverName=${artifactId}
+kieserver.restContextPath=/rest
+kieserver.location=http://localhost:${appServerPort}/rest/server
```
maybe replace localhost with ${appServerAddress}

mswiderski commented on kie-archetypes/kie-service-spring-boot-archetype/src/main/resources/META-INF/archetype-post-generate.groovy:
```diff
@@ -0,0 +1,357 @@
+launchFileContent = launchFileContent.replaceAll(myServiceNameMarker, myAppArtifactId);
+launchFileContent = launchFileContent.replaceAll(myServiceVersionMarker, myAppVersion);
+devLaunchFileContent = devLaunchFileContent.replaceAll(myServiceNameMarker, myAppArtifactId);
+devLaunchFileContent = devLaunchFileContent.replaceAll(myServiceVersionMarker, myAppVersion);
```
I believe this replace will generate incorrect setup
```
projects=( "test-service-model" "test-service-kjar" "test-service")
```
while the application name I created was test and thus the projects generated where like this
- test-model
- test-kjar
- test-service

so the launch script builds only test-service and skips the test-model and test-kjar

mswiderski commented on kie-archetypes/kie-service-spring-boot-archetype/src/main/resources/archetype-resources/launch-dev.sh:
```diff
@@ -0,0 +1,66 @@
+
+    "$mvnBin" -v
+    echo
+    projects=( "MYSERVICE_NAME_MARKER-model" "MYSERVICE_NAME_MARKER-kjar" "MYSERVICE_NAME_MARKER")
```
I think enough will be 

```
projects=( "*-model" "*-kjar" "*-service")
```

mswiderski commented on kie-archetypes/kie-service-spring-boot-archetype/src/main/resources/archetype-resources/launch-dev.sh:
```diff
@@ -0,0 +1,66 @@
+fi
+
+echo "Launching the application in development mode - requires connection to controller (workbench)"
+pattern="MYSERVICE_NAME_MARKER"
```
should be enough to use

```
pattern="*-service"
```

tsurdilo replied:
I think for archetype alone this cant be done (assumes users name app *-service"
*-kjar and *-model are ok I think. changed it for that

mswiderski commented on kie-archetypes/kie-service-spring-boot-archetype/src/main/resources/archetype-resources/launch-dev.sh:
```diff
@@ -0,0 +1,66 @@
+pattern="MYSERVICE_NAME_MARKER"
+files=( $pattern )
+cd ${files[0]}
+executable="$(ls  *target//MYSERVICE_NAME_MARKER-MYSERVICE_VERSION_MARKER.jar | tail -n1)"
```
better to use 
```
executable="$(ls  *target/*.jar | sort -V | tail -n1)"
```

as this will work regardless of version changes of the service project

tsurdilo replied:
This failed for me on OSX. Got error when running it so I removed it cause others can run into same issue.

mswiderski commented on kie-archetypes/kie-service-spring-boot-archetype/src/main/resources/archetype-resources/launch.sh:
```diff
@@ -0,0 +1,66 @@
+
+    "$mvnBin" -v
+    echo
+    projects=( "MYSERVICE_NAME_MARKER-model" "MYSERVICE_NAME_MARKER-kjar" "MYSERVICE_NAME_MARKER")
```
same changes as for the launch-dev.sh

tsurdilo commented:
@mswiderski updated with changes for all your comments except 1 (see my reply above).
Also added the kie-server-state xml we talked about - for this you have to add 3 more parameters when generating:

kjarGroupId
kjarArtifactId
kjarVersion

Let me know if there is anything else. Thanks.

tsurdilo commented:
@mswiderski with the addition of the kie-server-state xml file since we create it all the time (with default values with archetype only), when starting app it will throw error:

2018-08-09 12:11:00.949 ERROR 43908 --- [           main] o.k.server.services.impl.KieServerImpl   : Error creating container 'myserviceapp-kjar-1_0' for module 'org.kie.businessapp:myserviceapp-kjar:1.0'

The app starts OK still but this error will prompt users to think there is something wrong w/ their app setup.

If there something we can do about this ?

tsurdilo commented:
@mswiderski please note that unless you specify all 3 of these new kjar values when building archetype, the associated xml fill will NOT be built. This is because if users just use this archetype alone they do not have maybe kjar built yet, and in case we built this xml file w/ some default values, on service startup they will see for example:

2018-08-09 12:11:00.949 ERROR 43908 --- [           main] o.k.server.services.impl.KieServerImpl   : Error creating container 'myserviceapp-kjar-1_0' for module 'org.kie.businessapp:myserviceapp-kjar:1.0'

and might think the archetype or their built app has issues. That's why unless you specify the 3 kjar params the xml will not be generated . just fyi
